### ISSUE TITLE
Nested template literals behave strangely

### ISSUE THREAD

**Issue by beta-phenylethylamine** [mooz/js2-mode#520]
Noticed some strange behavior with indentation when nesting template literals. I imagine this is because inbuilt parsing is used to detect when inside strings for indentation, and this is causing it to understand the inside of the second template literal as the end of the first one and so on

```js
test_fun(test`
this works
${test`
	 this doesn't
	 ${test`
this does again
`}`}`);
```

**Comment by dgutov** [comment]
True. But please see the second item here: https://github.com/mooz/js2-mode#bugs

**Comment by beta-phenylethylamine** [comment]
@dgutov do template literals fall under the scope of js-mode though? i wasn't aware js-mode was intended to support es6 features

**Comment by dgutov** [comment]
That's a fair question. `js-mode` does support template literals to an extent, but maybe nested ones will give it some trouble.

We do rely on its indentation logic, though, so trying to support them there first would be natural.
